elementor/elementor
# Add Optimization Detective experiment

## PR Checklist



## PR Type
What kind of change does this PR introduce?


## Summary

This PR can be summarized in the following changelog entry:

* Add Optimization Detective experiment to leverage real user metrics to optimize the loading of images and embeds.

## Description
An explanation of what is done in this PR

* Create a new Optimization Detective experiment which embeds the [Optimization Detective](https://wordpress.org/plugins/optimization-detective/) plugin as well as the [Image Prioritizer](https://wordpress.org/plugins/image-prioritizer/) and [Embed Optimizer](https://wordpress.org/plugins/embed-optimizer/) plugins from [Performance Lab](https://wordpress.org/plugins/performance-lab/). Also installed, but not yet included, is the WIP plugin [Elementor LCP Background Image Preloader](https://github.com/westonruter/elementor-lcp-background-image-preloader) (which needs an Elementor engineer to assist with completing). These plugins are downloaded via Composer by means of [civicrm/composer-downloads-plugin](https://github.com/civicrm/composer-downloads-plugin).

## Test instructions
This PR can be tested by following these steps:

1. Activate the Optimization Detective experiment.
2. Log out, since Optimization Detective only optimizes pages for non-admins.
3. Navigate to a page that has one or more `IMG` elements.
4. While scrolled to the top of the page, reload in desktop, tablet, and mobile viewport sizes over the course of a few minutes. (By default, storage of new URL metrics locked once per minute for a given IP.)
5. The loading of the images should now be optimized, with `fetchpriority=high` and `loading=lazy` added to the correct `IMG` elements. Note also that a preload link with `fetchpriority=high` will be added with the appropriate media query. (Only if an `IMG` is the LCP element across all viewport sizes will it get the `fetchpriority=high` attribute.) The loading of Embeds should be similarly optimized, with embeds out of the initial viewport being lazy-loaded.

## Quality assurance
